File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    long credit;

    // Asks the user to input a credit card number.
    do
    {
        credit = get_long("Credit Card number: ");
    }
    // Checks if the user entered the credit card number correctly.
    while (credit <= 0);

    long check = credit;
    long digits = credit;
    int sum = 0;
    int count = 0;
    long divisor = 10;

    while (check > 0)
    {
        // Checks every first digit.
        int last = check % 10;
        sum += last;
        check /= 10;

        // Checks every second digit.
        last = check % 10;
        int x = last * 2;
        sum += (x % 10) + (x / 10);
        check /= 10;
    }
    // Checking amount of digits.
    while (digits > 0)
    {
        digits /= 10;
        count++;
    }
    // Finding the first digit.
    long first = credit;
    while(first >= 10)
    {
        first = first / 10;
    }
    // Finding the second digit.
    long first_two = credit;
    while(first_two >= 100)
    {
        first_two = first_two / 10;
    }

    int first_digit = first;
    int first_two_digits = first_two;

    // Final sumcheck.
    if (sum % 10 == 0)
    {
        // checks if VISA.
        if (first_digit == 4 && (count == 13 || count == 16))
        {
            printf("VISA\n");
        }
        // checks if American Express.
        else if ((first_two_digits == 34 || first_two_digits == 37) && count == 15)
        {
            printf("AMEX\n");
        }
        // checks if MasterCard.
        else if ((50 < first_two_digits && 56 > first_two_digits) && count == 16)
        {
            printf("MASTERCARD\n");
        }
        else
        // otherwise the card is invalid.
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: pset6 - Python 🐍/credit.py
from cs50 import get_int
from sys import exit
from math import floor

# Asks user for input the card value, it will repeat the question if the answer isn't a positive number
card = 0
while (card <= 0):
    card = get_int("Credit Card number: ")

check = card
last = 0
total = 0

while (check > 0):
    # Checks every first digit
    last = check % 10
    total += last

    # Gets rid of right most digit
    check /= 10
    check = floor(check)

    # Checks every second digit
    sec_last = (check % 10) * 2

    # If the number is 2 digits, it will add them both individually to the total
    if (sec_last >= 10):
        total += sec_last % 10
        sec_last /= 10
        total += floor(sec_last)
    else:
        total += sec_last

    # Checks every first digit
    check /= 10
    check = floor(check)

# Counts the amount of digits in the card
digits = card
count = len(str(digits))

# Finds the first digit on the card
first = card
while (first >= 10):
    first /= 10
first = int(first)

# Finds the first two digits on the card
first_two = card
while (first_two >= 100):
    first_two /= 10
first_two = int(first_two)

# Luhn’s algorithm check
if (total % 10 != 0):
    print("INVALID")
    exit(1)

# Check if the card is a VISA card
if (first == 4 and (count == 13 or count == 16)):
    print("VISA")
    exit(0)

# Checks if the card is an American Express card
elif ((first_two == 34 or first_two == 37) and count == 15):
    print("AMEX")
    exit(0)

# Checks if the card is a MasterCard
elif ((50 < first_two and 56 > first_two) and count == 16):
    print("MASTERCARD")
    exit(0)

# Otherwise the card is invalid
else:
    print("INVALID")
    exit(2)
